libvips/foreign/gifload.c

```diff
@@ -575,19 +572,17 @@ vips_foreign_load_gif_set_header( VipsForeignLoadGif *gif, VipsImage *image )
 		vips_image_set_int( image, 
 			VIPS_META_PAGE_HEIGHT, gif->file->SHeight );
 	vips_image_set_int( image, VIPS_META_N_PAGES, gif->n_pages );
-	vips_image_set_int( image, "gif-delay", gif->delay );
 	vips_image_set_int( image, "gif-loop", gif->loop );
-	if( gif->comment ) 
-		vips_image_set_string( image, "gif-comment", gif->comment );
 
-{
-	int array[10];
-	int i;
+  if ( gif->delays ) {
+	  vips_image_set_int( image, "gif-delay", VIPS_RINT( gif->delays[0] / 10.0 ) );
+	  vips_image_set_array_int( image, "delay", gif->delays, gif->n_pages );
+  } else {
+	  vips_image_set_int( image, "gif-delay", 4 );
+  }
```

Comment:
I'm not sure why it is aligned like this. Looks OK in editor 😞 

Maybe a TABS vs SPACES issue 🙂